src/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Sentry\Client;
+use Sentry\Options;
+
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var Options
+     */
+    protected $options;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkIdentifier;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkVersion;
+
+    public function __construct(Options $options, string $sdkIdentifier, string $sdkVersion)
+    {
+        $this->options = $options;
+        $this->sdkIdentifier = $sdkIdentifier;
+        $this->sdkVersion = $sdkVersion;
+    }
+
+    public function sendRequest(string $requestData): Response
+    {
+        $dsn = $this->options->getDsn();
+        if (null === $dsn) {
+            throw new \RuntimeException('The DSN option must be set to use the HttpClient.');
+        }
+
+        $curlHandle = curl_init();
+        curl_setopt($curlHandle, \CURLOPT_URL, $dsn->getEnvelopeApiEndpointUrl());
+        curl_setopt($curlHandle, \CURLOPT_HTTPHEADER, $this->getRequestHeaders());
+        curl_setopt($curlHandle, \CURLOPT_USERAGENT, $this->sdkIdentifier . '/' . $this->sdkVersion);
+        curl_setopt($curlHandle, \CURLOPT_TIMEOUT, $this->options->getHttpTimeout());
+        curl_setopt($curlHandle, \CURLOPT_CONNECTTIMEOUT, $this->options->getHttpConnectTimeout());
+        curl_setopt($curlHandle, \CURLOPT_SSL_VERIFYPEER, true);
+        curl_setopt($curlHandle, \CURLOPT_ENCODING, '');
```

Comment:
Isn't this the option to have compression? We may skip that in this PR (gzip caused issues in the past) but we may require it as a feature later.

Reply:
This flag only handles automatic decompression of HTTP downloads. Setting it to `''` enables all supported built-in compressions.